intel/init.py

```diff
@@ -172,6 +241,44 @@ def check_isolated_cores(platform, num_exclusive_cores, num_shared_cores):
                 (num_isolated_cores, required_isolated_cores))
 
 
+def check_sst_cp_isolated_cores(platform, num_exclusive_cores,
+                                num_shared_cores, epp_order):
+    error_occured = False
```

Comment:
nit: could we replace that flag with proper try-catch block and raising an Exception instead of assigning bool value? The purpose here is to capture all configuration errors, right? If so, I don't mind this :)

Reply:
Just looking through the code I'm a little confused as to what exactly to wrap in the try-catch block and what exception to raise, where do you suggest?

Reply:
below in this same function you have something like this:
```
if foo:
    error_occured = True # first error here, can raise an exception here and do sys.exit early
elif bar:
    error_occured = True # but instead you carry on and hit that one as well

if error_occured:
    sys.exit() # and now finally we exit the app anyway in a non-graceful way
```
instead you could replace all `error_occured = True` with `raise SomeException()` and do try-catch where you call this function:
```
try:
    check_sst_cp_isolated_cores(some, args, here)
catch SomeException() as e:
    log("blah blah")
    sys.exit(1)
```
hope you get the idea now

but if the intention is to go through all possible validations, then I'm okay with that as I mentioned in the initial comment

Reply:
Oh okay I understand what you mean. The idea behind carrying on after catching an error is to get all of the configuration errors in one run through.

Reply:
And that's totally fine, thanks, I'll mark as resolved.